### PR TITLE
PW-4462 Updated to checkout v67

### PR DESCRIPTION
**Pull request by saquibsayyad**
**Description**
Refactored 3DS1, 3DS2 and other HPP flow based on action response
Cleaned up unused old code.

**Tested scenarios**
Tested 3DS1, 3DS2, HPP flows including negative tests.

### CODE REVIEW

File: adyenv6core/src/com/adyen/v6/exceptions/AdyenNonAuthorizedPaymentResultException.java [hunk 1]
```diff
@@ -24,24 +24,24 @@
 import com.adyen.model.checkout.PaymentsResponse;
 import com.adyen.model.terminal.TerminalAPIResponse;
 
-public class AdyenNonAuthorizedPaymentException extends Exception {
```
Comment (rkewlani) [inline adyenv6core/src/com/adyen/v6/exceptions/AdyenNonAuthorizedPaymentResultException.java:27]: What is the reason to rename it?

File: adyenv6b2ccheckoutaddon/acceleratoraddon/web/webroot/WEB-INF/views/responsive/pages/checkout/multi/3ds_payment.jsp [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<%@ page trimDirectiveWhitespaces="true" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<html>
+<head>
+    <script src="https://${checkoutShopperHost}/checkoutshopper/sdk/4.3.1/adyen.js"
+            integrity="sha384-eNk32fgfYxvzNLyV19j4SLSHPQdLNR+iUS1t/D7rO4gwvbHrj6y77oJLZI7ikzBH"
```
Comment (rkewlani) [inline adyenv6b2ccheckoutaddon/acceleratoraddon/web/webroot/WEB-INF/views/responsive/pages/checkout/multi/3ds_payment.jsp:6]: do we need integrity?

Comment (saquibsayyad) [reply adyenv6b2ccheckoutaddon/acceleratoraddon/web/webroot/WEB-INF/views/responsive/pages/checkout/multi/3ds_payment.jsp:6]: it is recommended in our (docs)[https://docs.adyen.com/online-payments/components-web?tab=script_2#step-2-add-components] so i added it

File: adyenv6b2ccheckoutaddon/acceleratoraddon/web/webroot/WEB-INF/views/responsive/pages/checkout/multi/3ds_payment.jsp [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            crossorigin="anonymous"></script>
+    <link rel="stylesheet"
```
[the remark this answers is not in the document]
Comment (saquibsayyad) [reply adyenv6b2ccheckoutaddon/acceleratoraddon/web/webroot/WEB-INF/views/responsive/pages/checkout/multi/3ds_payment.jsp:8]: yes, i will update it

**Review by Aleffio — APPROVED** [review]

**Review by martinsrenato — APPROVED** [review]

File: adyenv6core/src/com/adyen/v6/exceptions/AdyenNonAuthorizedPaymentResultException.java (hunk 1 above)
Comment (rkewlani) [reply adyenv6core/src/com/adyen/v6/exceptions/AdyenNonAuthorizedPaymentResultException.java:27]: As discussed, this is an additional class to support paymentdetailsresponse type. AdyenNonAuthorizedPaymentResultException is already handling more than one types. Can you please check if we can use existing class?

Comment (saquibsayyad) [reply adyenv6core/src/com/adyen/v6/exceptions/AdyenNonAuthorizedPaymentResultException.java]: Added the field paymentsDetailsResponse in AdyenNonAuthorizedPaymentException

**Review by Aleffio — APPROVED** [review]

**Review by martinsrenato — APPROVED** [review]

**Review by rkewlani — APPROVED** [review]

**Pull request merged**